File: src/core/controller/KeyboardCore.java
package core.controller;

import core.controller.internals.AbstractKeyboardCoreImplementation;

/**
 * Class to provide API to control keyboard.
 * @author HP Truong
 *
 */
public class KeyboardCore extends AbstractKeyboardCoreImplementation {

	private final AbstractKeyboardCoreImplementation k;

	protected KeyboardCore(AbstractKeyboardCoreImplementation k) {
		this.k = k;
	}

	@Override
	public void type(String... strings) {
		k.type(strings);
	}

	@Override
	public void type(char... chars) {
		k.type(chars);
	}

	@Override
	public void type(int... keys) throws InterruptedException {
		k.type(keys);
	}

	@Override
	public void combination(int... keys) {
		k.combination(keys);
	}

	@Override
	public void hold(int key, int duration) throws InterruptedException {
		k.hold(key, duration);
	}

	@Override
	public void press(int... keys) {
		k.press(keys);
	}

	@Override
	public void release(int... keys) {
		k.release(keys);
	}

	/**
	 * Type a key multiple times
	 * @deprecated use {@link #repeat(int, int...)} instead
	 * @param key integer representing the key as specified in java.awt.events.KeyEvent class
	 * @param count number of times to repeat the typing
	 * @throws InterruptedException
	 */
	@Deprecated
	public void typeRepeat(int key, int count) throws InterruptedException {
		k.repeat(key, count);
	}

	@Override
	public boolean isLocked(int key) {
		return k.isLocked(key);
	}
}

File: src/natives/csharp/Core/KeyboardCore.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Repeat.core {
    public class KeyboardCore {
        public void type(string typing) {

        }

        public void type(char c) {

        }

        public void type(int key) {

        }

        public void type(params int[] keys) {

        }

        public void combination(params int[] keys) {

        }

        public void hold(int key, int duration) {

        }

        public void press(int key) {

        }

        public void press(params int[] keys) {
            foreach (int key in keys) {
                press(key);
            }
        }

        public void release(int key) {

        }

        public void release(params int[] keys) {
            foreach (int key in keys) {
                release(key);
            }
        }
    }
}
